django/contrib/auth/management/commands/createsuperuser.py

```diff
@@ -147,11 +156,15 @@ def handle(self, *args, **options):
 
                 user_data[self.UserModel.USERNAME_FIELD] = username
                 for field_name in self.UserModel.REQUIRED_FIELDS:
+                    value = None
```

Comment:
How about this?
```python
for field_name in self.UserModel.REQUIRED_FIELDS:
    value = options.get(field_name, os.environ.get('DJANGO_SUPERUSER_' + field_name.upper()))
    if value is None:
        raise CommandError('You must use --%s with --noinput.' % field_name)
    field = self.UserModel._meta.get_field(field_name)
    user_data[field_name] = field.clean(value, None)
```
Alternatively, a bit more verbose:
```python
for field_name in self.UserModel.REQUIRED_FIELDS:
    env_var = 'DJANGO_SUPERUSER_' + field_name.upper()
    value = os.environ.get(env_var)
    value = options.get(field_name, value)
    if value is None:
        raise CommandError('You must use --%s with --noinput.' % field_name)
    field = self.UserModel._meta.get_field(field_name)
    user_data[field_name] = field.clean(value, None)
```

Reply:
we can not use these two because if we don't provide options as command line params, by default `options` variable has it with `None` value. so the environment variable will never be used and this test will fail:
`test_use_environment_variable_non_interactive`

Reply:
:+1: grrr, presence of items with `None` vs non-existence. Thank for the nudge.